libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2934,7 +2934,14 @@ IRVariable IRGeneratorForStatements::readFromLValue(IRLValue const& _lvalue)
 			solUnimplementedAssert(_lvalue.type.isValueType(), "");
 			solUnimplementedAssert(_lvalue.type.sizeOnStack() == 1, "");
 			solAssert(_lvalue.type == *_immutable.variable->type(), "");
-			define(result) << "loadimmutable(\"" << to_string(_immutable.variable->id()) << "\")\n";
+			if (m_context.immutableRegistered(*_immutable.variable))
```

Comment:
I would prefer to have a "construction / deploy time" flag in the context, like we do in the old code generator.